examples/Azure/Example.ContainerAppJob/IHostApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using Azure.Monitor.OpenTelemetry.AspNetCore;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Resources.Azure;
+using OpenTelemetry.Trace;
+
+namespace Example.ContainerAppJob;
+
+public static class IHostApplicationBuilderExtensions
+{
+    public static IHostApplicationBuilder ConfigureOpenTelemetry(this IHostApplicationBuilder builder)
+    {
+        builder.AddAzureContainerAppsDetector();
+
+        builder.Logging.AddOpenTelemetry(logging =>
+        {
+            logging.IncludeFormattedMessage = true;
+            logging.IncludeScopes = true;
+        });
+
+        builder.Services.AddOpenTelemetry()
+            .WithMetrics(metrics =>
+             {
+                 metrics.AddAspNetCoreInstrumentation()
+                     .AddHttpClientInstrumentation()
+                     .AddRuntimeInstrumentation();
+             })
+            .WithTracing(tracing =>
+            {
+                tracing.AddSource("Example.ContainerAppJob")
+                    .AddAspNetCoreInstrumentation()
+
+                    // Uncomment the following line to enable gRPC instrumentation (requires the OpenTelemetry.Instrumentation.GrpcNetClient package)
+                    // .AddGrpcClientInstrumentation()
+                    .AddHttpClientInstrumentation();
+            });
+
+        builder.AddOpenTelemetryExporters();
+
+        return builder;
+    }
+
+    private static IHostApplicationBuilder AddAzureContainerAppsDetector(this IHostApplicationBuilder builder)
+    {
+        // Using AzureContainerAppsResourceDetector here requires adding the InternalsVisibleTo attribute in the OpenTelemetry.Resources.Azure package.
+        builder.Services.AddSingleton<AzureContainerAppsResourceDetector>();
+
+        builder.Services.AddOpenTelemetry()
+            .ConfigureResource(builder => builder
```

Comment:
Is there any reason why cannot use .`ConfigureResource(builder => builder.AddAzureContainerAppsResourceDetector())`?

It could remove all magic related to InternalsVisibleTo.

What is more, I think it is part which should not be included into documentation.

Reply:
> extension methods only work on a specific builder type.

I am not sure I follow that. ResourceDetectors provide an extension method on `this ResourceBuilder builder` only, and it works with both way of initializing SDK, be it SDK.Create* or builder.Services.OpenTelemetry() style! It is up to app owner to decide which way they should initialize SDK, but the person authoring ResourceDetector should be un-affected by that choice, as they *always* provide extension method on `this ResourceBuilder` only.

> From a developer perspective, I found it confusing and annoying at least 🤦🏼‍♂️.

I am sorry you feel that way! That was not the intend, and based on my above understanding (shared above), that is not the case either. But if there is some bug which prevent this from working, we can fix that. Also, If https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/trace/customizing-the-sdk/README.md#resource doc is not making this clear, I think we can add some wording to avoid any confusions there.

Reply:
Given example app is removed, resolving this conversation. Feel free to re-open if there are concerns/questions.

Reply:
I'm sorry, I only saw
- the `OpenTelemetryBuilder` returned from `builder.Services.AddOpenTelemetry()`
- the `TracingProviderBuilder` inside `OpenTelemetryBuilder.WithTracing(...)`
- the `MeterProviderBuilder` inside `OpenTelemetryBuilder.WithMetrics(...)`

I missed the point that `ConfigureResource(...)` (which does indeed give access to the `ResourceBuilder` needed for `AddAzureContainerAppsResourceDetector`) is usable on `builder.Services.AddOpenTelemetry()` as well and not only on Sdk.CreateTracerProviderBuilder() - my apologies.